hnn_core/gui/gui.py

```diff
@@ -490,10 +269,18 @@ def __init__(self, theme_color="#8A2BE2", log_window_height="100px",
         # Connectivity list
         self.connectivity_widgets = list()
 
-        # In-memory storage of all simulation related simulation_data
-        self.simulation_data = defaultdict(lambda: dict(net=None, dpls=list()))
-
         self._init_ui_components()
+        self.add_logging_window_logger()
+
+    def add_logging_window_logger(self):
+        # logger setting
+        global logger
+        logger = logging.getLogger(__name__)
```

Comment:
don't make it global. Put it at the top of the file, then import it from the other file. Like here:

https://github.com/mne-tools/mne-python/blob/858d1b25aafe98c54786e4344d2e5a11ac65da4b/mne/utils/_logging.py#L22-L23

We have to start using logging all over the repository (see https://github.com/jonescompneurolab/hnn-core/pull/388). It is outside the scope of your GSoC to implement that but it might be worth thinking how to keep it a little general so that future refactoring is not difficult.

cc @ntolley you might be interested in this issue as well